File: starrysky/src/main/java/com/lzx/starrysky/provider/MediaQueueProviderSurface.java
package com.lzx.starrysky.provider;

import android.graphics.Bitmap;
import android.support.v4.media.MediaBrowserCompat;
import android.support.v4.media.MediaMetadataCompat;

import com.lzx.starrysky.BaseMediaInfo;

import java.util.List;

/**
 * 数据提供类
 */
public class MediaQueueProviderSurface implements MediaQueueProvider {

    private MediaQueueProvider provider;

    public MediaQueueProviderSurface(MediaQueueProvider provider) {
        this.provider = provider;
    }

    @Override
    public List<BaseMediaInfo> getMediaList() {
        return provider.getMediaList();
    }

    @Override
    public List<SongInfo> getSongList() {
        return provider.getSongList();
    }

    @Override
    public void updateMediaList(List<BaseMediaInfo> mediaInfoList) {
        provider.updateMediaList(mediaInfoList);
    }

    @Override
    public void updateMediaListBySongInfo(List<SongInfo> songInfos) {
        provider.updateMediaListBySongInfo(songInfos);
    }

    @Override
    public void addMediaInfo(BaseMediaInfo mediaInfo) {
        provider.addMediaInfo(mediaInfo);
    }

    @Override
    public boolean hasMediaInfo(String songId) {
        return provider.hasMediaInfo(songId);
    }

    @Override
    public BaseMediaInfo getMediaInfo(String songId) {
        return provider.getMediaInfo(songId);
    }

    @Override
    public BaseMediaInfo getMediaInfo(int index) {
        return provider.getMediaInfo(index);
    }

    @Override
    public SongInfo getSongInfo(String songId) {
        return provider.getSongInfo(songId);
    }

    @Override
    public SongInfo getSongInfo(int index) {
        return provider.getSongInfo(index);
    }

    @Override
    public int getIndexByMediaId(String songId) {
        return provider.getIndexByMediaId(songId);
    }

    @Override
    public List<MediaMetadataCompat> getMediaMetadataCompatList() {
        return provider.getMediaMetadataCompatList();
    }

    @Override
    public List<MediaBrowserCompat.MediaItem> getChildrenResult() {
        return provider.getChildrenResult();
    }

    @Override
    public Iterable<MediaMetadataCompat> getShuffledMediaMetadataCompat() {
        return provider.getShuffledMediaMetadataCompat();
    }

    @Override
    public Iterable<BaseMediaInfo> getShuffledMediaInfo() {
        return provider.getShuffledMediaInfo();
    }

    @Override
    public MediaMetadataCompat getMediaMetadataCompatById(String songId) {
        return provider.getMediaMetadataCompatById(songId);
    }

    @Override
    public void updateMusicArt(String songId, MediaMetadataCompat changeData, Bitmap albumArt, Bitmap icon) {
        provider.updateMusicArt(songId, changeData, albumArt, icon);
    }

}

File: starrysky/src/main/java/com/lzx/starrysky/provider/MediaQueueProviderSurface.kt
package com.lzx.starrysky.provider

import android.graphics.Bitmap
import android.support.v4.media.MediaBrowserCompat
import android.support.v4.media.MediaMetadataCompat

import com.lzx.starrysky.BaseMediaInfo

/**
 * 数据提供类
 */
open class MediaQueueProviderSurface(private val provider: MediaQueueProvider) :
    MediaQueueProvider {
    override fun addMediaInfo(mediaInfo: BaseMediaInfo?) {
        provider.addMediaInfo(mediaInfo)
    }

    override fun getMediaList(): MutableList<BaseMediaInfo> {
        return provider.getMediaList()
    }

    override fun getSongList(): MutableList<SongInfo> {
        return provider.getSongList()
    }

    override fun getMediaInfo(songId: String): BaseMediaInfo? {
        return provider.getMediaInfo(songId)
    }

    override fun getMediaInfo(index: Int): BaseMediaInfo? {
        return provider.getMediaInfo(index)
    }

    override fun getSongInfo(songId: String): SongInfo? {
        return provider.getSongInfo(songId)
    }

    override fun getSongInfo(index: Int): SongInfo? {
        return provider.getSongInfo(index)
    }

    override val mediaMetadataCompatList: List<MediaMetadataCompat>
        get() = provider.mediaMetadataCompatList

    override val childrenResult: List<MediaBrowserCompat.MediaItem>
        get() = provider.childrenResult

    override val shuffledMediaMetadataCompat: Iterable<MediaMetadataCompat>
        get() = provider.shuffledMediaMetadataCompat

    override val shuffledMediaInfo: Iterable<BaseMediaInfo>
        get() = provider.shuffledMediaInfo

    override fun updateMediaList(mediaInfoList: List<BaseMediaInfo>) {
        provider.updateMediaList(mediaInfoList)
    }

    override fun onlyOneMediaBySongInfo(songInfo: SongInfo) {
        provider.onlyOneMediaBySongInfo(songInfo)
    }

    override fun updateMediaListBySongInfo(songInfos: List<SongInfo>) {
        provider.updateMediaListBySongInfo(songInfos)
    }

    override fun addMediaBySongInfo(info: SongInfo) {
        provider.addMediaBySongInfo(info)
    }

    override fun deleteMediaById(songId: String) {
        provider.deleteMediaById(songId)
    }

    override fun hasMediaInfo(songId: String): Boolean {
        return provider.hasMediaInfo(songId)
    }

    override fun getIndexByMediaId(songId: String): Int {
        return provider.getIndexByMediaId(songId)
    }

    override fun getMediaMetadataCompatById(songId: String?): MediaMetadataCompat? {
        return provider.getMediaMetadataCompatById(songId)
    }

    override fun updateMusicArt(
        songId: String, changeData: MediaMetadataCompat, albumArt: Bitmap, icon: Bitmap
    ) {
        provider.updateMusicArt(songId, changeData, albumArt, icon)
    }
}
